hpx/runtime/parcelset/parcelport_impl.hpp

```diff
@@ -753,6 +786,10 @@ namespace hpx { namespace parcelset
         /// The connection cache for sending connections
         util::connection_cache<connection, locality> connection_cache_;
 
+        typedef hpx::lcos::local::spinlock mutex_type;
+        mutex_type sender_threads_mtx_;
+        std::list<threads::thread_id_type> sender_threads_;
+
```

Comment:
A std::set might indeed be more appropriate. This way we can avoid one linear search. We need one linear search to scan for suspended threads though. The maximum number of active threads is limited by the maximum connections in the cache.